CRM/Contribute/Form/Contribution.php

```diff
@@ -1021,6 +1046,23 @@ public static function formRule($fields, $files, $self) {
       $errors['financial_type_id'] = ' ';
       $errors['_qf_default'] = $e->getMessage();
     }
+    if (!$self->_mode && !empty($fields['pan_truncation'])
+    ) {
+      $contributionDoneViaPaymentProcesor = FALSE;
```

Comment:
@pradpnayak please reply to @eileenmcnaughton indicating that you have accepted her point and will make a change or why you don't think it is necessary.